### PR TITLE
Fix build script to not fail on schedule changes

### PR DESCRIPTION
**Pull request by haakonvt**
From issue raised on slack regarding changes to a schedule causing the build script to fail creating the deploy matrix jobs:
https://cognitedata.slack.com/archives/C010VCGKBCM/p1680267530939639

This also bumps the Python version used by default in the template to `3.9`. Sweet.

### CODE REVIEW

File: build_scripts/changed_functions.py [hunk 1]
```diff
@@ -14,42 +14,43 @@ def parse_args():
...
-    if not deploy:
-        deploy = ["skipDeploy"]
-
-    json_format = json.dumps({"folders": deploy})
+        deploy_all_folder = args.deploy_all.strip()
+        print(f"Input: Common folder (may force deploy all): {deploy_all_folder!r}")
+
+    # Compare against previous commit under the assumption of squash-only merges:
+    diff = subprocess.check_output("git diff --name-only HEAD^^ HEAD".split(), text=True).split()
+    changed_files = set(map(Path, diff))
```
Comment (haakonvt) [inline build_scripts/changed_functions.py:33]:
```suggestion
    diff = subprocess.check_output("git diff --name-only HEAD^ HEAD".split(), text=True).split()
    changed_files = set(map(Path, diff))
```

File: build_scripts/changed_functions.py [hunk 2]
```diff
@@ -14,42 +14,43 @@ def parse_args():
...
-            deploy = list(function_folders)
-            break
-        if changed_folder in function_folders:
-            deploy.append(changed_folder)
-
-    if not deploy:
-        deploy = ["skipDeploy"]
-
-    json_format = json.dumps({"folders": deploy})
+        deploy_all_folder = args.deploy_all.strip()
```
Comment (doctrino) [inline build_scripts/changed_functions.py:28]: You are currently expecting only one folder that can trigger deploy all? In the previous version, I think I allowed for multiple common folders. 

I think I agree with this restriction. Any extra validation needed?

File: build_scripts/changed_functions.py [hunk 3]
```diff
@@ -14,42 +14,43 @@ def parse_args():
...
+        print(f"Input: Common folder (may force deploy all): {deploy_all_folder!r}")
+
+    # Compare against previous commit under the assumption of squash-only merges:
+    diff = subprocess.check_output("git diff --name-only HEAD^ HEAD".split(), text=True).split()
+    changed_files = set(map(Path, diff))
```
Comment (doctrino) [inline build_scripts/changed_functions.py:33]: It always take me a few seconds extra to read `map`, more pythonic to use a comprehension?
```suggestion
    changed_files = {Path(path) for path in diff}
```

**Review by doctrino — APPROVED** [review]
Remember to also update the CHANGELOG before merging. 

Just some minor comments.

File: build_scripts/changed_functions.py (hunk 2 above)
Comment (haakonvt) [reply build_scripts/changed_functions.py:28]: The action only supports a single common folder, so this should be the default I think.

File: build_scripts/changed_functions.py (hunk 3 above)
Comment (haakonvt) [reply build_scripts/changed_functions.py:33]: 👀

**Comment by haakonvt** [comment]
@doctrino TY for review, I had it on auto-merge, but will update changelog in a new PR 👍

**Pull request merged**